doc/source/modulefile.rst

```diff
@@ -97,344 +95,306 @@ the *modulefile* is being loaded.
  enabled in case no argument is provided. If a list contains more than one
  *directory*, then each member acts as a boolean OR operation.
 
-.. _is-avail:
+.. modulefilecmd:: is-avail modulefile...
 
-**is-avail** modulefile...
-
- The **is-avail** command returns a true value if any of the listed
- *modulefiles* exists in enabled **MODULEPATH**. If a list contains more than
+ The :modulefilecmd:`is-avail` command returns a true value if any of the listed
+ *modulefiles* exists in enabled :envvar:`MODULEPATH`. If a list contains more than
  one *modulefile*, then each member acts as a boolean OR operation. If an
- argument for **is-avail** is a directory and a *modulefile* exists in the
- directory **is-avail** would return a true value.
+ argument for :modulefilecmd:`is-avail` is a directory and a *modulefile* exists in the
+ directory :modulefilecmd:`is-avail` would return a true value.
 
  The parameter *modulefile* may also be a symbolic modulefile name or a
  modulefile alias. It may also leverage a specific syntax to finely select
  module version (see `Advanced module version specifiers`_ section below).
 
-.. _module:
-
-**module** [sub-command] [sub-command-args]
+.. modulefilecmd:: module [sub-command] [sub-command-args]
 
  Contains the same *sub-commands* as described in the :ref:`module(1)`
  man page in the Module Sub-Commands section. This command permits a
- *modulefile* to **load** or **unload** other *modulefiles*. No checks are
+ *modulefile* to :modulefilecmd:`load` or :modulefilecmd:`unload` other *modulefiles*. No checks are
  made to ensure that the *modulefile* does not try to load itself. Often
  it is useful to have a single *modulefile* that performs a number of
- **module load** commands. For example, if every user on the system
+ ``module load`` commands. For example, if every user on the system
  requires a basic set of applications loaded, then a core *modulefile*
- would contain the necessary **module load** commands.
+ would contain the necessary ``module load`` commands.
 
- Command line switches **--auto**, **--no-auto** and **--force** are ignored
- when passed to a **module** command set in a *modulefile*.
+ Command line switches :option:`--auto`, :option:`--no-auto` and :option:`--force` are ignored
+ when passed to a :command:`module` command set in a *modulefile*.
 
-.. _module-info:
+.. modulefilecmd:: module-info option [info-args]
 
-**module-info** option [info-args]
-
- Provide information about the **modulecmd.tcl** program's state. Some of the
- information is specific to the internals of **modulecmd.tcl**. *option*
+ Provide information about the :file:`modulecmd.tcl` program's state. Some of the
+ information is specific to the internals of :file:`modulecmd.tcl`. *option*
  is the type of information to be provided, and *info-args* are any
  arguments needed.
 
- **module-info type**
+ .. modulefilecmd:: module-info type
```

Comment:
should only declare top modulefilecmd `module-info`, not for each sub-command of this modulefile command

Reply:
Done.